cloudinit/net/networkd.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+# vi: ts=4 expandtab
+#
+# Copyright (C) 2021 VMware Inc.
+#
+# Author: Shreenidhi Shedi <yesshedi@gmail.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import os
+
+from . import renderer
+from cloudinit import util
+from cloudinit import subp
+from cloudinit import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class CfgParser:
+    def __init__(self):
+        self.conf_dict = {
+            'Match': [],
+            'Link': [],
+            'Network': [],
+            'DHCPv4': [],
+            'DHCPv6': [],
+            'Address': [],
+            'Route': [],
+        }
+
+    def update_section(self, sec, key, val):
+        for k in self.conf_dict.keys():
+            if k == sec:
+                self.conf_dict[k].append(key+'='+str(val))
+
+    def get_final_conf(self):
+        contents = ''
+        for k, v in self.conf_dict.items():
+            if not v:
+                continue
+            contents += '['+k+']\n'
+            for e in v:
+                contents += e + '\n'
+            contents += '\n'
+
+        return contents
+
+    def dump_data(self, target_fn):
+        if not target_fn:
+            LOG.warning('Target file not given')
+            return
+
+        contents = self.get_final_conf()
+        LOG.debug('Final content: %s', contents)
+        util.write_file(target_fn, contents)
+
+
+class Renderer(renderer.Renderer):
+    ''' Renders network information in /etc/systemd/network '''
```

Comment:
Can we add a comment or docstring here specifying that this Renderer is currently experimental and doesn't support all the use cases supported by the other renderers yet.